SE-I-T-Digital/device-opcua-go
# Kamakura Update

# PR Checklist

> **If your build fails** due to your commit message not passing the build checks, please review the guidelines [here](https://github.com/edgexfoundry/device-sdk-go/blob/main/.github/Contributing.md).

Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change
- [x] I am not introducing a new dependency
- [x] I have added unit tests for the new feature
- [x] I have fully tested the new feature

## Breaking Changes

- Removed custom configuration; added parameters to device protocol attributes

## Testing Instructions

- Start Prosys Simulation Server
- Update `Endpoint` in `/cmd/res/devices/devices.toml`
- Optionally, define more `[[DeviceList]]` entries in `/cmd/res/devices/devices.toml`
- Optionally, set `WRITABLE_LOGLEVEL` to `"DEBUG"` and tail the device-opcua-go Docker log to see readings data being published to core-data
- Add/Remove/Update devices and device profiles using the [core-metadata API](https://app.swaggerhub.com/apis/EdgeXFoundry1/core-metadata/2.2.0)
- Query core-data to confirm subscription readings have been added
- Use the [device-sdk API](https://app.swaggerhub.com/apis/EdgeXFoundry1/device-sdk/2.2.0) to read and write data of specific resources defined in the device profiles
